scripts/bootloader/zb_add_ota_header.py

```diff
@@ -118,7 +118,7 @@ def convert_version_string_to_int(s):
     numbers = s.split(".")
     if len(numbers) != 3:
         raise ValueError("application-version-string parameter must be on the format x.y.z")
-    js = [1024*1024, 1024, 0]
+    js = [0x100*0x10000, 0x10000, 1]
```

Comment:
format will be like 10020003, comment above is now not relevant

Reply:
That's correct. Zigbee specification reserves 4 bytes for describing the OTA File Version. We wanted to reuse the version provided by MCUBoot to avoid confision.

The MCUBOOT versioning is defined as:
```
struct mcuboot_img_sem_ver {
	uint8_t major;
	uint8_t minor;
	uint16_t revision;
	uint32_t build_num;
};
```
So the most logical way of using it was to construct the Zigbee image version as (in hexadecimal notation):
```
<1-byte-major><1-byte-minor><2-bytes-revision>
```
As the result, is you get the version number provided as string: `<major>.<minor>.<patch>` and want to convert it into numeric representation, the `<major>` and `<minor>` values has to be shifted by 2 additional hexadecimal digit (thus `0x10000` instead of `0x100`).

The comment string is now aligned with the scheme presented above.